presto-hive/src/main/java/com/facebook/presto/hive/OrcFileWriter.java

```diff
@@ -245,14 +246,18 @@ public Optional<Page> commit()
         try {
             orcWriter.close();
         }
-        catch (IOException | UncheckedIOException e) {
+        catch (IOException | UncheckedIOException | PrestoException e) {
             try {
                 rollbackAction.call();
             }
             catch (Exception ignored) {
                 // ignore
             }
-            throw new PrestoException(HIVE_WRITER_CLOSE_ERROR, "Error committing write to Hive. " + e.getMessage(), e);
+            ErrorCodeSupplier errorCodeSupplier = HIVE_WRITER_CLOSE_ERROR;
+            if (e instanceof PrestoException) {
+                errorCodeSupplier = () -> ((PrestoException) e).getErrorCode();
+            }
+            throw new PrestoException(errorCodeSupplier, "Error committing write to Hive. " + e.getMessage(), e);
```

Comment:
wondering why we need to wrap a PrestoException inside another PrestoException in line 260.?
If it is a PrestoException do we need to even catch it, since looks like all we are doing is adding an additional message to the exception.

If you feel we need to catch it then can we keep that in a separate catch block above the current one so that you can get rid of the instanceOf check in line 257. 

Also the rollbackAction should move into the finally block since it is a cleanup similar to how it is done in  rollback() method below. 
Do we need the same error code handling done in orcWriter.close() call in rollback() as well.

Reply:
Yeah after looking more ..the extra wrapping is not necessary. Let me streamline it and just propagate it if its a presto exception

Reply:
This is a commit call. we need to rollback only if there is an exception during writer.close() call.

Reply:
I was talking about the rollback code a few lines below which also uses  orcWriter.close();
 ```
    @Override
    public void rollback()
    {
        try {
            try {
                orcWriter.close();
            }
```

Reply:
Discussed offline with Ajay, we need to rollback only when writer.close fails . So no need to put rollbackAction.call in final block.